File: python/dfml/comment.py
import element

class Comment(element.Element):
	"""
	Represents a comment element.

	"""
	def create(string: str = "") -> "Comment":
		"""
		Factory method to create a Comment instance.

		Args:
			string (str): The comment string.

		Returns:
			Comment: A Comment instance.
		"""
		return Comment(string)

	def __init__(self, string: str) -> None:
		"""
		Constructor for Comment.

		Args:
			string (str): The comment string.
		"""
		self.__string = string

	def get_element_type(self) -> int:
		"""
		Get the element type (COMMENT).

		Returns:
			int: The element type.
		"""
		return element.Element.COMMENT

	def get_string(self) -> str:
		"""
		Get the comment string.

		Returns:
			str: The comment string.
		"""
		return self.__string

	def set_string(self, string: str) -> None:
		"""
		Set the comment string.

		Args:
			string (str): The new comment string.
		"""
		self.__string = string

File: js/main/comment.js
import { DFMLElement } from "./element.js"

/**
 * Class representing a comment element in the Dragonfly Markup Language (DFML).
 *
 * @export
 * @class DFMLComment
 * @extends {DFMLElement}
 */
export class DFMLComment extends DFMLElement {
	/**
	 * Creates an instance of DFMLComment.
	 * @param {string} [content=""] content text of the comment.
	 * @memberof DFMLComment
	 */
	constructor(content = "") {
		super();
		this.content = content;
	}
  
	/**
	 * Creates and returns a shared pointer to an empty DFMLComment instance.
	 *
	 * @static
	 * @return {DFMLComment} the new DFMLComment instance.
	 * @memberof DFMLComment
	 */
	static create() {
		return new DFMLComment();
	}

	/**
	 * Creates and returns a DFMLComment instance with the specified string content.
	 *
	 * @static
	 * @param {string} content the content of the comment.
	 * @return {*} the new DFMLComment instance.
	 * @memberof DFMLComment
	 */
	static createWithContent(content) {
		const comment = this.create(); 
		comment.setString(content); 
		return comment; 
	}
  
	/**
	 * Sets the string content of the comment.
	 *
	 * @param {string} content The content to set for the comment.
	 * @memberof DFMLComment
	 */
	setString(content) {
		this.content = content;
	}
  
	/**
	 * Gets the string content of the comment.
	 *
	 * @return {string} The content of the comment.
	 * @memberof DFMLComment
	 */
	getString() {
		return this.content;
	}
  
	/**
	 * Gets the element type as an integer, identifying it as a comment.
	 *
	 * @return {number} 
	 * @override
	 * @memberof DFMLComment The element type (COMMENT).
	 */
	getElementType() {
		return DFMLElement.COMMENT;
	}
  }
